### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by AronBuzogany**
This PR solves #8 with 11 tests currently. 

It also adds a pre test configuration file `src/setupTests.js` due to an issue discussed in  - https://github.com/inrupt/solid-client-authn-js/issues/1676

### CODE REVIEW

**Comment by pheyvaer** [comment]
Did you also have a look at https://github.com/SolidLabResearch/solid-web-app-template/ for the testing?

**Comment by AronBuzogany** [comment]
> Did you also have a look at https://github.com/SolidLabResearch/solid-web-app-template/ for the testing?

I will do this now.

**Comment by AronBuzogany** [comment]
I have added a test for login authentication. 
I will also add tests for error handling once #24  is merged

File: cypress/e2e/spec.cy.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        cy.contains('button', 'Log in').click();
+        cy.contains('button', 'Authorize').click();
+
+        cy.contains('Logged in as:')
+
+        cy.contains('A list of my favorite books').click()  
+        cy.contains('"It Ends With Us"')      
+    })
+
+    it("Query on private data unauthenticated", () => {
```
Comment (pheyvaer) [inline cypress/e2e/spec.cy.js:22]: Can you also add a test for public data?

**Review by pheyvaer — CHANGES_REQUESTED** [review]
Starting the app gives the following warning:
```
WARNING in [eslint] 
src/typeMapper.js
  Line 4:23:  img elements must have an alt prop, either with meaningful text, or an empty string for decorative images  jsx-a11y/alt-text
```

Running the tests gives the following error:

```
 PASS  src/__tests__/RightField.test.jsx
  ● Console

    console.error
      Warning: Invalid DOM property `for`. Did you mean `htmlFor`?
          at label
          at form
          at div
          at SolidLoginForm (/Users/pheyvaer/Development/solidlab/generic-data-viewer/src/components/SolidLoginForm.jsx:9:35)
          at div
          at div
          at RightField (/Users/pheyvaer/Development/solidlab/generic-data-viewer/src/right-field/RightField.jsx:33:31)
```

File: README.md [hunk 2]
```diff
@@ -117,3 +117,32 @@ To create a local pod with which you can test for example authentication you can
   We already added a query to list books based on the resource `favourite-books` to `src/config.json`.
 - Log in with the IDP `http://localhost:8080` and 
   the credentials in the file `seeded-pod-config.json`. 
+
+
+## Testing 
+
+For testing we use [Cypress](https://www.cypress.io/) or [React-Jest](https://jestjs.io/docs/tutorial-react).
```
[the remark this answers is not in the document]
Comment (AronBuzogany) [reply README.md:124]: I was thinking that they don't depend on each other, so we either write a test in Cypress or a test in React-Jest

**Comment by AronBuzogany** [comment]
> Starting the app gives the following warning:

I have made a [separate issue](https://github.com/SolidLabResearch/generic-data-viewer/issues/37) for this as it is not related to this PR

**Comment by AronBuzogany** [comment]
I have made a [separate issue](https://github.com/SolidLabResearch/generic-data-viewer/issues/38) for this one also

File: README.md [hunk 3]
```diff
@@ -117,3 +117,32 @@ To create a local pod with which you can test for example authentication you can
...
+To run all the available tests you can execute the following:
```
[the remark this answers is not in the document]
Comment (AronBuzogany) [reply README.md]: I have resolved this in [commit 17740df](https://github.com/SolidLabResearch/generic-data-viewer/pull/28/commits/17740df88f62ad4dfd7733fec7f8934599f61ce9)

File: README.md [hunk 4]
```diff
@@ -117,3 +117,32 @@ To create a local pod with which you can test for example authentication you can
...
+
+1. Prepare and start the community Solid server with the available pods as explained in the [Testing with local pods section](#testing-with-local-pods).
```
[the remark this answers is not in the document]
Comment (AronBuzogany) [reply README.md]: This is corrected in [commit 7b8964b](https://github.com/SolidLabResearch/generic-data-viewer/pull/28/commits/7b8964bdca58cad2ef5e82bb7417246a14cafe8c)

File: src/__tests__/ExtandableSelector.test.jsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { act, render, screen } from "@testing-library/react"
+import userEvent from '@testing-library/user-event'
+import '@testing-library/jest-dom'
+import ExtendableSelector from "../components/ExtendableSelector"
+
+
+
+
```
[the remark this answers is not in the document]
Comment (AronBuzogany) [reply src/__tests__/ExtandableSelector.test.jsx]: Removed unnecessary lines in [commit f97040c](https://github.com/SolidLabResearch/generic-data-viewer/pull/28/commits/f97040cd44ae125b766c1c93999fa391f23cf3ff)

File: src/__tests__/RightField.test.jsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import config from "../config.json";
+import { render } from "@testing-library/react"
+import '@testing-library/jest-dom'
+import RightField from "../right-field/RightField";
+import { TextEncoder, TextDecoder } from 'util';
+
+Object.assign(global, { TextDecoder, TextEncoder });
+
+
```
[the remark this answers is not in the document]
Comment (AronBuzogany) [reply src/__tests__/RightField.test.jsx]: Removed unnecessary lines in [commit f97040c](https://github.com/SolidLabResearch/generic-data-viewer/pull/28/commits/f97040cd44ae125b766c1c93999fa391f23cf3ff)

File: src/__tests__/RightField.test.jsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+if (!config.queryFolder) {
+    config.queryFolder = "./";
+}
+
+
```
[the remark this answers is not in the document]
Comment (AronBuzogany) [reply src/__tests__/RightField.test.jsx]: Removed unnecessary lines in [commit f97040c](https://github.com/SolidLabResearch/generic-data-viewer/pull/28/commits/f97040cd44ae125b766c1c93999fa391f23cf3ff)

File: src/__tests__/SelectionTable.test.jsx [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import { render } from "@testing-library/react"
+import userEvent from '@testing-library/user-event'
+import '@testing-library/jest-dom'
+import SelectionTable from "../selection-table/SelectionTable"
+
+
+
```
[the remark this answers is not in the document]
Comment (AronBuzogany) [reply src/__tests__/SelectionTable.test.jsx]: Removed unnecessary lines in [commit f97040c](https://github.com/SolidLabResearch/generic-data-viewer/pull/28/commits/f97040cd44ae125b766c1c93999fa391f23cf3ff)

File: src/__tests__/ResultsTable.test.jsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    expect(noQueryLabel).not.toBeInTheDocument()
+})
+
+test('A query with variables exposes the gridjs table', () => {
+    const {container} = render(<ResultsTable variables={["Test1"]} results={["testData"]} selectedQuery={{name: "Test query"}}></ResultsTable>)
+    const gridJsTable = container.getElementsByClassName('gridjs').item(0)
+    expect(gridJsTable).not.toBeUndefined()  
+})
+
+test('The GridJS table isn\'t present unless there are variables', () => {
```
[the remark this answers is not in the document]
Comment (AronBuzogany) [reply src/__tests__/ResultsTable.test.jsx]: I have adjusted this in [commit 8f2ae6a](https://github.com/SolidLabResearch/generic-data-viewer/pull/28/commits/8f2ae6a856265de6d23131e480e68bfc06965191)

File: src/__tests__/ResultsTable.test.jsx [hunk 10]
```diff
@@ -0,0 +1,30 @@
+import {render, screen } from "@testing-library/react"
```
[the remark this answers is not in the document]
Comment (AronBuzogany) [reply src/__tests__/ResultsTable.test.jsx]: I have adjusted this in [commit f0ad0b9](https://github.com/SolidLabResearch/generic-data-viewer/pull/28/commits/f0ad0b9208da7ab3037fc8c868a64e97c92048e4)

File: initial-pod-data/wish-list$.jsonld
[the remark this answers is not in the document]
Comment (AronBuzogany) [reply initial-pod-data/wish-list$.jsonld:1]: I have added a public data access test for this in [commit bcb449f](https://github.com/SolidLabResearch/generic-data-viewer/pull/28/commits/bcb449f7d7405b935ab49d3a88eaac3fe4f30f83)

File: cypress/e2e/spec.cy.js (hunk 1 above)
Comment (AronBuzogany) [reply cypress/e2e/spec.cy.js:22]: As of [commit bcb449f](https://github.com/SolidLabResearch/generic-data-viewer/pull/28/commits/bcb449f7d7405b935ab49d3a88eaac3fe4f30f83) this test is available.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (pheyvaer) [reply README.md:124]: Then I would write "we use X and Y. We use X for .... We use Y for ..."

[the remark this answers is not in the document]
Comment (AronBuzogany) [reply README.md]: Agreed, I have adjusted it respectively in [commit 135c073](https://github.com/SolidLabResearch/generic-data-viewer/pull/28/commits/135c073b8d8dd55ba62cfe63df8c8464dbfe1a66)

**Review by pheyvaer — APPROVED** [review]

**Pull request merged**